src/Microsoft.CodeDom.Providers.DotNetCompilerPlatform/Util/CompilationSettings.cs

```diff
@@ -54,5 +80,14 @@ internal static class CompilationSettingsHelper {
             string compilerFullPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, relativePath);
             return compilerFullPath;
         }
+
+        private static bool IsDebuggerAttached {
+            get {
+                return IsDebuggerPresent() || Debugger.IsAttached;
+            }
+        }
+
+        [DllImport("kernel32.dll")]
```

Comment:
Because it's now importing kernel32.dll, it won't work on Mono/Linux.
Throws System.EntryPointNotFoundException
Mono 5.0.1.1 on Ubuntu Server 16.04